src/node/mod.rs

```diff
@@ -61,22 +64,41 @@ pub async fn node_with_autodiscovery(addresses_override: Option<String>) -> anyh
     Ok(())
 }
 
-fn parse_socket_list(socket_list: String) -> Vec<SocketAddr> {
-    socket_list
-        .split(',')
-        .map(|s| SocketAddr::from_str(s).unwrap())
-        .collect::<Vec<SocketAddr>>()
+fn parse_socket_list(socket_list: Vec<String>) -> Result<Vec<SocketAddr>, anyhow::Error> {
+    let (sockets, errors): (Vec<Result<_, _>>, Vec<Result<_, _>>) = socket_list
+        .iter()
+        .map(|s| SocketAddr::from_str(s))
+        .partition(Result::is_ok);
+    if errors.is_empty() {
+        let addresses = sockets
+            .iter()
+            .map(|x| *x.as_ref().unwrap())
+            .collect::<Vec<SocketAddr>>();
+        Ok(addresses)
+    } else {
+        Err(anyhow!("Could not parse correctly all addresses"))
+    }
 }
 
 #[cfg(test)]
 mod tests {
     use super::*;
 
     #[test]
-    fn parse_socket_list_test() {
-        let list = "1.2.3.4:4321,5.4.3.2:9876";
-        let sockets = parse_socket_list(list.to_string());
+    fn parse_socket_list_test() -> Result<(), anyhow::Error> {
+        let list = vec!["1.2.3.4:4321".to_string(), "5.4.3.2:9876".to_string()];
+        let sockets = parse_socket_list(list)?;
         let socket_under_test = sockets[1];
-        assert!(socket_under_test.is_ipv4() && socket_under_test.port() == 9876)
+        assert!(socket_under_test.is_ipv4() && socket_under_test.port() == 9876);
+        Ok(())
+    }
+
+    #[test]
+    fn parse_socket_list_should_fail_when_one_address_is_malformed() {
```

Comment:
Good tests! 👍 

Just a small tip, you can:
```rust
assert!(parse_socket_list(list).is_err(), "should fail because of missing port")
```
without having to do a whole match with multiple asserts.